### PR TITLE
feat: Full screen sidekick library

### PR DESCRIPTION
**Pull request by dylandepass**
https://github.com/adobe/franklin-sidekick-library/assets/3231084/77d9ec64-670f-4324-be6d-aa76f11a7288

This PR changes the rendering of the sidekick library into a full screen experience.

## Notable changes
* Application now runs on same origin as the content. Customers will create a `library.html` file in their repository at `/tools/sidekick/library.html`. This static page will load the sidekick library from `hlx.live`.
* In context editing that allows text and image content to be edited within the sidekick library and then copied into the document.
* More web components from spectrum are now available
* Plugins to the library are now rendered in picker located in the header
* Support for `library-metadata` block for authoring block names and descriptions
* Local development now runs on HTTPS

## Breaking changes from last version
* Each variation of a block in the document must be separated by a section delimiter.
* Custom plugins may need to be adjusted to work in the larger screen space.

## Known Issues
* Sites with authentication enable will be sent to the IDP login screen if there `hlx-auth-token` cookie is not set. After login they will be redirected to the root of the site and not back to the sidekick library. An [enhancement](https://github.com/adobe/helix-admin/issues/1293) is required to helix-admin to fix this.
* Currently Drag and Dropping images and then hitting copy may cause the tab to crash. There is very little to go on to debug this and more time will need to spent to figure it out. The work around for now is to [disable Hardware acceleration](https://www.howtogeek.com/412738/how-to-turn-hardware-acceleration-on-and-off-in-chrome/) in chrome.

### CODE REVIEW

**Comment by dylandepass** [comment]
@rofe Will fix this flaky test in the morning ;)

File: index.html [hunk 1]
```diff
@@ -4,7 +4,7 @@
 <head>
   <meta charset="utf-8">
   <meta name="viewport" content="width=device-width, initial-scale=1.0, viewport-fit=cover" />
-  <meta name="Description" content="Put your description here.">
+  <meta name="Description" content="Sidekick Library">
```
Comment (rofe) [inline index.html:7]:
```suggestion
  <meta name="Description" content="AEM Sidekick Library">
```

File: README.md [hunk 2]
```diff
@@ -1,16 +1,16 @@
 # Franklin Sidekick Library
```
Comment (rofe) [inline README.md:1]:
```suggestion
# AEM Sidekick Library
```

File: README.md [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 
-[DEMO](https://main--boilerplate-with-library--dylandepass.hlx.page/sidekick/library?suppressFrame=true)
+[DEMO](https://main--boilerplate-with-library--dylandepass.hlx.live/tools/sidekick/library.html)
 
 ## What is the Sidekick Library?
 
-The Sidekick Library is a plugin for the Franklin Sidekick that allows developers to build UI based tooling for their franklin content authoring enviroment (office, gdocs). It comes standard with a blocks plugin that can display a list of all blocks to authors in an intuitive manner, removing the need for authors to remember or search for every variation of a block. Developers can also write their own plugins for the library or even write their own blocks plugin.
+The Sidekick Library is an extension for the Franklin Sidekick that enables developers to create UI-driven tooling for franklin content authors. It includes a built-in blocks plugin that can display a list of all blocks to authors in an intuitive manner, removing the need for authors to remember or search for every variation of a block. Developers can also write their own plugins for the sidekick library.
```
Comment (rofe) [inline README.md:13]:
```suggestion
The Sidekick Library is an extension for the AEM Sidekick that enables developers to create UI-driven tooling for content authors. It includes a built-in blocks plugin that can display a list of all blocks to authors in an intuitive manner, removing the need for authors to remember or search for every variation of a block. Developers can also write their own plugins for the sidekick library.
```

File: README.md [hunk 4]
```diff
@@ -49,24 +51,116 @@ The Sidekick library comes with a `blocks` plugin.
...
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta
+      name="viewport"
+      content="width=device-width, initial-scale=1.0, viewport-fit=cover"
+    />
+    <meta name="Description" content="Franklin Sidekick Library" />
```
Comment (rofe) [inline README.md:83]:
```suggestion
    <meta name="Description" content="AEM Sidekick Library" />
```

File: certs/localhost-key.pem
Comment (rofe) [inline certs/localhost-key.pem:1]: Should this really be here? Feels like this should be an ENV variable in CircleCI...

File: certs/localhost.pem
Comment (rofe) [inline certs/localhost.pem:1]: Should this really be here? Feels like this should be an ENV variable in CircleCI...

File: src/locales/en/messages.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "appTitle": {
-    "message": "Libraries"
+    "message": "Sidekick Library"
```
Comment (rofe) [inline src/locales/en/messages.json:3]:
```suggestion
    "message": "AEM Sidekick Library"
```

**Review by rofe — APPROVED** [review]

File: index.html [hunk 6]
```diff
@@ -14,17 +14,18 @@
...
 </head>
 
 <body>
-  <script type="module" src="./src/app.js"></script>
+  <script type="module" src="./src/index.js"></script>
   <script>
     const library = document.createElement('sidekick-library');
     library.config = {
-      base: 'https://main--helix-test-content-onedrive--adobe.hlx.page/block-library-tests/library-single-sheet.json'
+      base: 'https://main--boilerplate-with-library--dylandepass.hlx.page/tools/sidekick/library.json',
```
Comment (tripodsan) [inline index.html:28]: shouldn't we move this to a public location?

**Review by tripodsan — CHANGES_REQUESTED** [review]
- remove localhost keys - even if they are only for development/testing, we don't want the github security scanners to complain about them
- did you vet the OpenAI usage with the openai legal team at adobe? it's a complicated process. if not, I would remove/disable the generative text support for now

File: index.html (hunk 6 above)
Comment (dylandepass) [reply index.html:28]: Hey @tripodsan, this html file is only used for development and is not deployed to `hlx.live`. Customer will created their own `library.html` file which will be equivalent to this one.

**Pull request merged**